docs/UserGuide.md

```diff
@@ -358,10 +359,21 @@ Examples:
   `clear`<br>
   `undo` (reverses the `clear` command)<br>
   `undo` (reverses the `cancel 1` command)
+  
+Step by step illustration:
+1. Initial state:
+![UndoCommandInitialState](images/UndoRedoInitialState_UG.png)
+2. Input `delete 1`:
+![UndoCommandStep2](images/UndoStep2_UG.png)
+The first patient `Alex Yeoh` is deleted.
+3. Input `Undo`:
+![UndoCommandStep3](images/UndoStep3_UG.png)
+The `undo` command cancels the last undoable command `delete 1`, so that 
+the patient `Alex Yeoh` comes back to the list again.
 
 ### Redoing the previously undone command : `redo`
 
-Reverses the most recent `undo` command.
+Reverses the most recent `undo` command. `redo` command will not work if you restart Nuudle.
```

Comment:
Same as above. This can be rephrased to:
`redo` command cannot reverse an `undo` execution from the previous Nuudle session.